packages/dds/map/src/mapKernel.ts

```diff
@@ -661,9 +673,15 @@ export class MapKernel {
                         0x2fd /* pendingMessageId does not match */);
                     this.submitMapClearMessage(op, localOpMetadata.previousMap);
                 },
-                getStashedOpLocalMetadata: (op: IMapClearOperation) => {
+                processStashedOpLocalMetadata: (op: IMapClearOperation) => {
+                    if (this.pendingKeys.size !== 0) {
```

Comment:
This if block should not be here. The stashed op is a local clear, so it is correct for it to get rid of the local key set edits.